source/extensions/transport_sockets/tls/ssl_socket.h

```diff
@@ -72,6 +72,7 @@ class SslSocket : public Network::TransportSocket,
   Network::TransportSocketCallbacks* transportSocketCallbacks() override { return callbacks_; }
 
 protected:
+  friend class SslSocketFriend;
```

Comment:
This class doesn't exist, I don't think? Delete.

Reply:
It's used in `handshaker_factory_test.cc` as a way to peek at an `SslSocket`'s underlying `SSL*`. I thought briefly about just making the `rawSsl()` accessor public instead of protected but decided that it would expose the internal `SSL*` too much.

Reply:
The pattern used elsewhere for this is to create a method `rawSslForTest()`.